### PR TITLE
BUG: None converted to NaN after groupby first and last

### LINKED ISSUE #32800: Groupby converts None to NaN
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd

df = pd.DataFrame.from_dict({
    'id': ['a'],
    'value': [None]
})

df_grouped = df.groupby('id', as_index=False).first()

print(df)
print(df_grouped)
```

Output:
```python
  id value
0  a  None
  id  value
0  a  NaN
```

#### Problem description

Since at least version 1.0.2,  the type of `df_grouped`  is `NaN`.  In Version 0.25.3, the type was `None`.
This breaks my code since I later check for this value using `if var is None`, which is False when `var` is `NaN` instead of `None`.

#### Expected Output

```python
  id value
0  a  None
  id  value
0  a  None
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.1.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-88-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8
pandas           : 1.0.2
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.0.0
Cython           : None
pytest           : 5.4.1
hypothesis       : None
sphinx           : 2.4.4
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.2.0
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : 5.4.1
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.15
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by JDkuba**
- [x] closes #32800
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

This PR solves #32800. In #32124 `not checknull()` was introduced to `groupfirst()` and `grouplast()`. `None` is not caught in this conditional and it's being converted to `nan` because of that.

I'm a newbie and not sure whether I should add whatsnew entry or not. Should I?

### CODE REVIEW

File: pandas/_libs/groupby.pyx [hunk 1]
```diff
@@ -893,7 +893,7 @@ def group_last(rank_t[:, :] out,
             for j in range(K):
                 val = values[i, j]
 
-                if not checknull(val):
+                if not checknull(val) or val is None:
```
Comment (jreback) [inline pandas/_libs/groupby.pyx:896]: can you add a comment here of why we are doing this

File: pandas/_libs/groupby.pyx [hunk 2]
```diff
@@ -986,7 +986,7 @@ def group_nth(rank_t[:, :] out,
...
```
Comment (jreback) [inline pandas/_libs/groupby.pyx:989]: same

File: pandas/tests/groupby/test_nth.py [hunk 3]
```diff
@@ -94,6 +94,16 @@ def test_nth_with_na_object(index, nulls_fixture):
     tm.assert_frame_equal(result, expected)
 
 
+@pytest.mark.parametrize("method", ["first", "last"])
+def test_first_last_with_None(method):
+    # https://github.com/pandas-dev/pandas/issues/32800
```
Comment (jreback) [inline pandas/tests/groupby/test_nth.py:99]: can you add a comment, that we wish to preserve None in object dtypes.

**Review by jreback — CHANGES_REQUESTED** [review]
looks fine, can you add a whatsnew in 1.1, groupby bug fixes. merge master and ping on green.

File: pandas/_libs/groupby.pyx (hunk 1 above)
Comment (JDkuba) [reply pandas/_libs/groupby.pyx:898]: Done

File: pandas/_libs/groupby.pyx (hunk 2 above)
Comment (JDkuba) [reply pandas/_libs/groupby.pyx:993]: Done

File: pandas/tests/groupby/test_nth.py (hunk 3 above)
Comment (JDkuba) [reply pandas/tests/groupby/test_nth.py:99]: Done

**Review by WillAyd — APPROVED** [review]
lgtm

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**